Mathlib/Analysis/NormedSpace/ProdLp.lean

```diff
@@ -0,0 +1,775 @@
+    Isometry (WithLp.equiv ∞ (α × β)) :=
+  fun x y =>
+  le_antisymm (by simpa only [ENNReal.coe_one, one_mul] using prod_lipschitzWith_equiv ∞ α β x y)
+    (by
+      simpa only [ENNReal.div_top, ENNReal.zero_toReal, NNReal.rpow_zero, ENNReal.coe_one,
+        one_mul] using prod_antilipschitzWith_equiv ∞ α β x y)
+
+/-- Seminormed group instance on the product of two normed groups, using the `L^p`
+norm. -/
+instance instProdSeminormedAddCommGroup [SeminormedAddCommGroup α] [SeminormedAddCommGroup β] :
+    SeminormedAddCommGroup (WithLp p (α × β)) where
+  dist_eq x y := by
+    rcases p.dichotomy with (rfl | h)
+    · simp only [prod_dist_eq_sup, prod_norm_eq_sup, dist_eq_norm]
+      rfl
+    · have : p ≠ ∞ := by
+        intro hp
+        rw [hp, ENNReal.top_toReal] at h
+        linarith
+      simp only [prod_dist_eq_add (zero_lt_one.trans_le h),
+        prod_norm_eq_add (zero_lt_one.trans_le h), dist_eq_norm]
+      rfl
+
+/-- normed group instance on the product of two normed groups, using the `L^p` norm. -/
+instance instProdNormedAddCommGroup [NormedAddCommGroup α] [NormedAddCommGroup β] :
+    NormedAddCommGroup (WithLp p (α × β)) :=
+  { instProdSeminormedAddCommGroup p α β with
+    eq_of_dist_eq_zero := eq_of_dist_eq_zero }
+
+example [NormedAddCommGroup α] [NormedAddCommGroup β] :
+    (instProdNormedAddCommGroup p α β).toMetricSpace.toUniformSpace.toTopologicalSpace =
+    instProdTopologicalSpace p α β :=
+  rfl
+
+example [NormedAddCommGroup α] [NormedAddCommGroup β] :
+    (instProdNormedAddCommGroup p α β).toMetricSpace.toUniformSpace = instProdUniformSpace p α β :=
+  rfl
+
+example [NormedAddCommGroup α] [NormedAddCommGroup β] :
+    (instProdNormedAddCommGroup p α β).toMetricSpace.toBornology = instProdBornology p α β :=
+  rfl
+
+section norm_of
+
+variable {p α β}
+variable [SeminormedAddCommGroup α] [SeminormedAddCommGroup β]
+
+theorem prod_nnnorm_eq_add (hp : p ≠ ∞) (f : WithLp p (α × β)) :
+    ‖f‖₊ = (‖f.fst‖₊ ^ p.toReal + ‖f.snd‖₊ ^ p.toReal) ^ (1 / p.toReal) := by
+  ext
+  simp [prod_norm_eq_add (p.toReal_pos_iff_ne_top.mpr hp)]
+
+theorem prod_nnnorm_eq_sup (f : WithLp ∞ (α × β)) : ‖f‖₊ = ‖f.fst‖₊ ⊔  ‖f.snd‖₊ := by
+  ext
+  norm_cast
+
+theorem prod_norm_eq_of_nat (n : ℕ) (h : p = n) (f : WithLp p (α × β)) :
+    ‖f‖ = (‖f.fst‖ ^ n + ‖f.snd‖ ^ n) ^ (1 / (n : ℝ)) := by
+  have := p.toReal_pos_iff_ne_top.mpr (ne_of_eq_of_ne h <| ENNReal.nat_ne_top n)
+  simp only [one_div, h, Real.rpow_nat_cast, ENNReal.toReal_nat, eq_self_iff_true, Finset.sum_congr,
+    prod_norm_eq_add this]
+
+theorem prod_norm_eq_of_L2 (x : WithLp 2 (α × β)) : ‖x‖ = sqrt (‖x.fst‖ ^ 2 + ‖x.snd‖ ^ 2) := by
+  rw [prod_norm_eq_of_nat 2 (by norm_cast) _, Real.sqrt_eq_rpow]
+  norm_cast
+
+theorem prod_nnnorm_eq_of_L2 (x : WithLp 2 (α × β)) :
+    ‖x‖₊ = NNReal.sqrt (‖x.fst‖₊ ^ 2 + ‖x.snd‖₊ ^ 2) :=
+  NNReal.eq <| by
+    push_cast
+    exact prod_norm_eq_of_L2 x
+
+variable (α β)
```

Comment:
I don't see why you would make the variables explicit here while they are implicit in all nearby lemmas, which are of a similar nature. Note that making variables explicit is less useful in Lean 4, because we can always resort to named arguments if needed.